mlir/include/mlir/Dialect/Tosa/IR/TosaTypesBase.td

```diff
@@ -74,16 +74,25 @@ def Tosa_QuantizedInt : AnyTypeOf<[Tosa_QuantizedType<"uint8", [8], 0>,
                                    Tosa_QuantizedType<"int16", [16, 0], 1>,
                                    Tosa_QuantizedType<"int32", [32, 0], 1>]>;
 
+def Tosa_F8 : AnyTypeOf<[
+                        F8E4M3FN,
+                        F8E5M2]>;
+
 //===----------------------------------------------------------------------===//
 // Multi-category types.
 //===----------------------------------------------------------------------===//
 def Tosa_AnyNumber : AnyTypeOf<[Tosa_Int, Tosa_QuantizedInt, AnyFloat],
                                 "number">;
 
+// Add F8 type support to Tosa_AnyNumber
+def Tosa_AnyNumber_Extended : AnyTypeOf<[Tosa_AnyNumber, Tosa_F8],
```

Comment:
Why do we need the extended type here?

Reply:
The goal here is to `differentiate between operators that take FP8 and those that do not take FP8`. 

Another way is to do the following: 

- define `Tosa_AnyNumber` to include every dtypes. 
- define something like `Tosa_AnyNumber_Exclude_FP8` to remove FP8 from that set. 

But this is no better than the existing solution.

Reply:
`Tosa_AnyNumber` already uses `AnyFloat` so I believe it could already support FP8?

Reply:
Ah, right. We didn't have this a year ago.

Reply:
updated.